analyzers/tests/SonarAnalyzer.Test/Rules/UseConstantLoggingTemplateTest.cs

```diff
@@ -0,0 +1,158 @@
+public class UseConstantLoggingTemplateTest
+{
+    private readonly VerifierBuilder builder = CreateVerifier<CS.UseConstantLoggingTemplate>();
+
+    [TestMethod]
+    public void UseConstantLoggingTemplate_CS() =>
+        builder.AddPaths("UseConstantLoggingTemplate.cs").Verify();
+
+    [DataTestMethod]
+    [DataRow("Log", "LogLevel.Warning,")]
+    [DataRow("LogCritical")]
+    [DataRow("LogDebug")]
+    [DataRow("LogError")]
+    [DataRow("LogInformation")]
+    [DataRow("LogTrace")]
+    [DataRow("LogWarning")]
+    public void UseConstantLoggingTemplate_MicrosoftExtensionsLogging_CS(string methodName, string extraParameter = "") =>
+        builder.AddSnippet($$"""
+            using Microsoft.Extensions.Logging;
+
+            public class Program
+            {
+                public void Method(ILogger logger, int arg)
+                {
+                    logger.{{methodName}}({{extraParameter}} "Message");    // Compliant
+                    logger.{{methodName}}({{extraParameter}} $"{arg}");     // Noncompliant
+                }
+            }
+            """).Verify();
+
+    [DataTestMethod]
+    [DataRow("Debug")]
+    [DataRow("Error")]
+    [DataRow("Fatal")]
+    [DataRow("Info")]
+    [DataRow("Trace")]
+    [DataRow("Warn")]
+    public void UseConstantLoggingTemplate_CastleCoreLogging_CS(string methodName) =>
+        builder.AddSnippet($$"""
+            using Castle.Core.Logging;
+
+            public class Program
+            {
+                public void Method(ILogger logger, int arg)
+                {
+                    logger.{{methodName}}("Message");            // Compliant
+                    logger.{{methodName}}($"{arg}");             // Noncompliant
+                    logger.{{methodName}}Format("{Arg}", arg);   // Compliant
+                    logger.{{methodName}}Format($"{arg}");       // Noncompliant
+                }
+            }
+            """).Verify();
+
+    [DataTestMethod]
+    [DataRow("Debug")]
+    [DataRow("Error")]
+    [DataRow("Fatal")]
+    [DataRow("Info")]
+    [DataRow("Warn")]
+    public void UseConstantLoggingTemplate_Log4Net_CS(string methodName) =>
+        builder.AddSnippet($$"""
+            using log4net;
+
+            public class Program
+            {
+                public void Method(ILog logger, int arg)
+                {
+                    logger.{{methodName}}("Message");               // Compliant
+                    logger.{{methodName}}($"{arg}");                // Noncompliant
+                    logger.{{methodName}}Format("Arg: {0}", arg);   // Compliant
+                    logger.{{methodName}}Format($"{arg}");          // Noncompliant
+                }
+            }
+            """).Verify();
+
+    [DataTestMethod]
+    [DataRow("ConditionalDebug")]
+    [DataRow("ConditionalTrace")]
+    [DataRow("Debug")]
+    [DataRow("Error")]
+    [DataRow("Fatal")]
+    [DataRow("Info")]
+    [DataRow("Trace")]
+    [DataRow("Warn")]
```

Comment:
Do we want to add support also for the logging methods from [IloggerBase](https://nlog-project.org/documentation/v5.0.0/html/Methods_T_NLog_ILoggerBase.htm)?